### PR TITLE
Fix coverity issues

### PR DESCRIPTION
**Pull request by vkalintiris**
##### Summary

Assorted changes to fix coverity issues.

##### Test Plan

CI + new coverity report.

##### Additional Information

Resolves https://github.com/netdata/netdata/issues/12596

### CODE REVIEW

File: daemon/unit_test.c [hunk 1]
```diff
@@ -420,6 +420,7 @@ int unit_test_static_threads() {
 
     if (n < 2) {
         fprintf(stderr, "only %d static threads registered", n);
+        free(static_threads);
```
Comment (MrZammler) [inline daemon/unit_test.c:423]: Hi Vasili! Not a big, but any reason not to use `freez` ?

Comment (vkalintiris) [reply daemon/unit_test.c]: Changed in latest commit.

**Review by MrZammler — APPROVED** [review]

File: ml/Host.h [hunk 2]
```diff
@@ -84,8 +84,8 @@ class TrainableHost : public RrdHost {
     std::pair<Dimension *, Duration<double>> findDimensionToTrain(const TimePoint &NowTP);
     void trainDimension(Dimension *D, const TimePoint &NowTP);
 
+    struct rusage ResourceUsage{};
```
Comment (erdem2000) [inline ml/Host.h:87]: I did not get the point about replacement of lines, just wondering the reason if the order effects something.

**Review by erdem2000 — APPROVED** [review]
Hİ Vasili, I could not see specific test steps so just compiled and executed the PR.  It works well.

**Pull request merged**